cmake/win32.cmake

```diff
@@ -25,8 +66,47 @@ if(NOT MSVC_VERSION)
   add_definitions(-DWINVER=0x0500 -D_WIN32_WINNT=0x0500)
   # Wait a minute, if we're not Microsoft C++, nor a Clang paired with Microsoft C++,
   # then the only possible option has to be GNU or a GNU-linked Clang!
-  if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0 OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    set(FS_LIB stdc++fs)
+  set(FS_LIB stdc++fs)
+  # If we're building a lokinet for windows, but need to target ancient hardware  
+  function(check_working_cxx_atomics64 varname)
```

Comment:
this seems like we should hoist this function to its own file, and not duplicate between all the different flavours here.